	<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <meta name="description" content="Benchmark Datasets for Bioinformatic Applications">
    <meta name="Keywords" content="biobench-sets, Biomolecules, Folding, Disease, Protein,
    Emidio, Capriotti, Variant, Mutation, Predict, RNA, Structure, Machine Learning,
    Bioinformatics, University, Bologna, SNP, genome, interpretation">  
    <meta name="author" content="BioBench Sets - Emidio Capriotti">
    <link rel="icon" href="/favicon.ico">

    <title>ThermoProt-DDG: Testing the prediction of DDG upon protein mutation</title>

    <!-- Bootstrap core CSS -->
    <link href="/pages/dist/css/bootstrap.min.css" rel="stylesheet">

    <!-- IE10 viewport hack for Surface/desktop Windows 8 bug 
    <link href="pages/assets/css/ie10-viewport-bug-workaround.css" rel="stylesheet">
    -->

    <!-- Custom styles for this template -->
    <link href="/pages/style/jumbotron-narrow-biofold.css" rel="stylesheet">

    <!-- Just for debugging purposes. Don't actually copy these 2 lines! -->
    <!--[if lt IE 9]><script src="../assets/js/ie8-responsive-file-warning.js"></script><![endif]
    <script src="pages/assets/js/ie-emulation-modes-warning.js"></script>
    -->

    
	   
  </head>

  <body >
    <br>
    <div class="container">

      <div class="jumbotron">
        <table width="100%">
        <tr>
        <td width="90%">
        <div class='biofold' ><span style="color:#c31e27;"></span><span style="color:#4689a3;">ProtDDG-Bench</span>
	<hr style="height:16pt; visibility:hidden;" />
        <span style="color:#4689a3;font-weight:300;font-size:16px;">Benchmarking resource for predictors of protein stability change</span>
        </div>
	</td>
	<td width="10%">
	<div class="right_cell">
        <a href="http://thermoprot-ddg.githum.io">
        <img src="/pages/img/biobench.png" width="100px" align="left">
        </a>
        </div>
	</td>
	</tr>
	</table>

	<table width="100%">
        <tr>
        <td>
        <div class="form-inline justify-content-center">
        <nav>
          <ul class="nav nav-pills">
            <li role="presentation" class="active" ><a href="#">Home</a></li>
	    <li role="presentation" ><a href="/vb1432">VB1432</a></li>
	    <li role="presentation" ><a href="/s2648">S2648</a></li>
	    <li role="presentation" ><a href="/ssym">SSYM</a></li>
	    <li role="presentation" ><a href="/broom">Broom</a></li>
	    <li role="presentation" ><a href="/myoglobin">Myoglobin</a></li>
	    <li role="presentation" ><a href="/p53">P53</a></li>
	    <li role="presentation" ><a href="/ptmul">PTMUL</a></li>
            <li role="presentation" ><a href="https://github.com/thermoprot-ddg/thermoprot-ddg">GitHub</a></li>
          </ul>
        </nav>
        </div>
        </td>
        </tr>
        </table>
        <br>
  	<br>
  	<div class="text">
     <h4 class="title" ></h4>
     </div>	
          <p class="text-small" >
		All the datasets for testing the performance of the predictors of DDG upon mutation are available at
		the <a href="https://github.com/thermoprot-ddg/thermoprot-ddg" target=blank>thermoprot-ddg</a> repository.
		The thermoprot-ddg repository includes the following datasets:
		<br><br>
		<ul>
                <li>
		<p class="text-small" >
   	        <a href="/vb1432">VB1432</a>:    
                1432 variants from 79 protein structures corresponding to 65 clusters.
                9 mutations have double experimental data. 1 mutation is not mapping
                to the strcuture. The 1LRP structure was replace with 1LMB. For structure 1WQ5
                was considered the chain B. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/29597263" target=blank>PMID:29597263</a>.
	        </li>
  		<li>
		<p class="text-small" >
                <a href="/s2648">S2648</a>:     
                2648 variants from 132 protein strcutures corresponding to 113 clusters.
                Experimental DDGs of the same variants are avereged. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/21569468" target=blank>PMID:21569468</a>.
                </li>    
		<li>
		<p class="text-small" >
                <a href="/ssym">Ssym</a>:      
                634 variants from 357 structures corresponding to 13 clusters.
                Dataset composed by 342 mutations and their reverse. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/29718106" target=blank>PMID:29718106</a>.
		</li>
   		<li>
		<p class="text-small" >
		<a href="/broom">Broom</a>:     
                Dataset composed by 605 mutations from 58 structures corresponding to 50 clusters.
                This dataset contains 53 mutations from non-native proteins and 59 mutations
                referring to fragment of the protein. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/28710274" target=blank>PMID:28710274</a>.
		</li>
   		<li>
		<p class="text-small" >
		<a href="/myoglobin">Myoglobin</a>: 
                134 variants from myoglobin from structure 1BZ6. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/26054434" target=blank>PMID:26054434</a>.
		</li>
   		<li>
		<p class="text-small" >
		<a href="/p53">P53</a>:       
		42 variants from P53 structure 2OCJ. Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/24281696" target=blank>PMID:24281696</a>.
		</li>
		<li>
		<p class="text-small" >
   		<a href="/ptmul">PTMUL</a>:     
		914 multiple site variants from 91 protein structures and 77 clusters.
                Data from 
		<a href="https://pubmed.ncbi.nlm.nih.gov/31266447" target=blank>PMID:31266447</a>.
		</li>
		</ul>
	<br>

	<div class="text">
	<h4 class="title" >Datasets files</h4>
	</div>

	<p class="text-small" >
	The directory S2648 and VB1432 contains 10 files for 10-folds cross-validation tests.
	Furthermore, the cross-validation subset of S2648 and VB1432 are consistent. This means that
  	the following predictions can be performed:
	<br>
	<br>
        <pre style="text-align: justify;">
	Training: not SET_i vb1432-10fold-split-j.tsv -> Test: SET_i s2648-10fold-split-j.tsv
        Training: not SET_i s2648-10fold-split-j.tsv  -> Test: SET_i vb1432-10fold-split-j.tsv</pre>
        <br>

	<p class="text-small" >
        The directory BROOM contains a 5-fold split of the BROOM dataset. Given the numeber of mutations
        mutations form the same cluster the set has been diveded in 5 subsets.
        The test on this dataset can be performed as follow:
	<br>
        <br>
        <pre style="text-align: justify;">
        Training: not SET_i train-vb1432-test-broom.tsv -> Test: SET_i broom-5fold.tsv
        Training: not SET_i train-s2648-test-broom.tsv  -> Test: SET_i broom-5fold.tsv</pre>
        <br>

	<p class="text-small" >
	The directory SSYM contains a 5-fold split of the Ssym dataset. Given the large number of
	mutations form the same cluster the set has been diveded in 5 subsets.
  	The test on this dataset can be performed as follow:
        <br>
        <br>
        <pre style="text-align: justify;">
         Training: not SET_i train-vb1432-test-ssym.tsv -> Test: SET_i ssym-5fold.tsv
         Training: not SET_i train-s2648-test-ssym.tsv  -> Test: SET_i ssym-5fold.tsv</pre>
	<br>
    
	<p class="text-small" >
	The directory MYOGLOBIN test contains the testing dataset myoglogin.tsv with the
  	best subsets of VB1432 and S2648 to be used as possible training.
  	The following prediction can be performed:
        <br>
        <br>
        <pre style="text-align: justify;">
        Training: train-vb1432-test-myoglobin.tsv (1399) -> Test: myoglobin.tsv
        Training: train-s2648-test-myoglobin.tsv  (2607) -> Test: myoglobin.tsv</pre>
	<br>

        <p class="text-small" >
	The directory P53 test contains the testing dataset p53.tsv with the best subsets
  	of VB1432 and S2648 to be used as possible training.
  	The following prediction can be performed:
	<br>
        <br>
        <pre style="text-align: justify;">
        Training: train-vb1432-test-p53.tsv       (1427) -> Test: p53.tsv
        Training: train-s2648-test-p53.tsv        (2643) -> Test: p53.tsv</pre>
	<br>

        <p class="text-small" >
	This directory contains files for testing predictions on multiple site mutations starting from
	a training on a set single point mutations.
  	The directory PTMUL contains a 5-fold split of the PTMUL dataset. Given the numeber of mutations
  	mutations form the same cluster the set has been diveded in 5 subsets.
  	The test on this dataset can be performed as follow:
	<br>
        <br>
	<pre style="text-align: justify;">
         Training: not SET_i train-vb1432-test-ptmul.tsv -> Test: SET_i ptmul-5fold.tsv
         Training: not SET_i train-s2648-test-ptmul.tsv  -> Test: SET_i ptmul-5fold.tsv</pre>
        <br>

	<div class="text">
	<h4 class="title" >Clustering</h4>
	</div>

        <p class="text-small" >
	The file <a href="https://github.com/thermoprot-ddg/thermoprot-ddg/blob/master/data/cluster-545-pdbchains.txt" target=blank>cluster-545-pdbchains.txt</a> 
	contains 132 clusters of 545 PDB chains.
	The clustering is obtained using blastclust with the options -S 25 -L 0.5 -b F.
	<br>
	<br>

	<div class="text">
	<h4 class="title" >Testing</h4>
	</div>

        <p class="text-small" >
	To test your method you need to:
	<p class="text-small" >
	1. replace the file 
        <a href="https://github.com/thermoprot-ddg/thermoprot-ddg/blob/master/scripts/predict-ddg-value.py">predict-ddg-value.py</a> 
        with your own script that runs taking
     	in input only the testing and training files and returning in standard output
     	the experimental and the predicted DDGs respectively
	The program runs as follows:
	<br>
        <br>
        <pre style="text-align: justify;">
        predict-ddg-value.py test_file.txt train_file.txt</pre>
	<br>
	<p class="text-small" >
	2. Generate an inputfile containing a two columns representing the PDB chain
	identifier and the mutation followed by all the inputfeatures.
	The full list of mutations are reported in the file data/unique-mutations-input.txt
     	and example of input file with two input features is 
	<a href="https://github.com/thermoprot-ddg/thermoprot-ddg/blob/master/data/ifeatures-KYTJ820101-BASU050101.txt" target=blank>ifeatures-KYTJ820101-BASU050101.txt</a>.
  	Finally run ./test.py input_feature_file.txt to score the performace of your method.
	For example runs:
	<br>
        <br>
        <pre style="text-align: justify;">
	./test.py data/ifeatures-KYTJ820101-BASU050101.txt</pre>
  <br>
  <br>
  <br>
    <div class='footer-ok'><div>

      </div>
    </div> <!-- /container -->


    <!-- IE10 viewport hack for Surface/desktop Windows 8 bug
    <script src="pages/assets/js/ie10-viewport-bug-workaround.js"></script>
    -->
  </body>
</html>
